cookbooks/magnetic_stripes/magnetic_stripes.cc

```diff
@@ -0,0 +1,119 @@
+#include <aspect/material_model/composition_reaction.h>
+#include <aspect/simulator_access.h>
+#include <aspect/global.h>
+#include <aspect/geometry_model/interface.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    template <int dim>
+    class MagneticStripes : public MaterialModel::CompositionReaction<dim>
+    {
+      public:
+
+        virtual void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+                              MaterialModel::MaterialModelOutputs<dim> &out) const;
+
+        static void declare_parameters (ParameterHandler &prm);
+        virtual void parse_parameters (ParameterHandler &prm);
+
+      private:
+        std::vector<double> reversal_times;
+    };
+
+
+    template <int dim>
+    void
+    MagneticStripes<dim>::
+    evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+             MaterialModel::MaterialModelOutputs<dim> &out) const
+    {
+      CompositionReaction<dim>::evaluate(in, out);
+
+      int magnetic_orientation = 1;
+      for (unsigned int i=0; i < reversal_times.size(); ++i)
+        if (this->get_time() < reversal_times[i])
+          {
+            magnetic_orientation = -2 * (i % 2) + 1;
```

Comment:
Add documentation about what this equation does? Does it simply produce alternating positive/negative values with each iteration?